torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -641,6 +647,7 @@ void TensorPipeAgent::sendCompletedResponseMessage(
 
     for (const auto& tensor : responseMessage->tensors()) {
       const auto device = tensor.device();
+      std::unique_lock<std::mutex> lock(groupMembershipMutex_);
```

Comment:
why do we need this lock here? and why this is inside the for loop? Since this is on the critical path for every RPC response, will this cause any perf issues?

Reply:
Thanks for catching. This was meant to also have the if statement check to only run for dynamic rpc, added the lock guard to this as well